clients/src/main/java/org/apache/kafka/common/requests/FetchRequest.java

```diff
@@ -153,7 +189,9 @@
     private final int minBytes;
     private final int maxBytes;
     private final IsolationLevel isolationLevel;
-    private final LinkedHashMap<TopicPartition, PartitionData> fetchData;
+    private final Map<TopicPartition, PartitionData> fetchData;
```

Comment:
Since the caller always passes in a LinkedHashMap, is there a reason to change this to Map?

Reply:
The rationale is that `FetchSessionHandler` wraps the `LinkedHashMap` in an `UnmodifiableMap`, so the type is technically `UnmodifiableMap` rather than `LinkedHashMap` at that point.  Also, there's things like using `Collections.emptyMap` in a unit test which you can't use if you have to have a `LinkedHashMap`

I will add a comment saying that iteration order is significant, though.